packages/fxa-auth-server/lib/routes/account.ts

```diff
@@ -1650,6 +1723,47 @@ export const accountRoutes = (
       },
       handler: (request: AuthRequest) => accountHandler.finishSetup(request),
     },
+    {
+      method: 'POST',
+      path: '/account/set_password',
+      options: {
+        ...MISC_DOCS.ACCOUNT_SET_PASSWORD_POST,
+        auth: {
+          mode: 'required',
+          payload: false,
+          strategy: 'oauthToken',
+        },
+        validate: {
+          query: isA.object({
+            sendVerifyEmail: isA
+              .boolean()
+              .optional()
+              .default(true)
+              .description(DESCRIPTION.sendVerifyEmail),
+          }),
+          payload: isA.object({
+            authPW: validators.authPW.description(DESCRIPTION.authPW),
+            metricsContext: METRICS_CONTEXT_SCHEMA,
```

Comment:
I'm requiring `metricsContext` here, though I plan to finalize where this comes from (e.g. do we already have it stashed in Redis with a `productId`?) in [FXA-5557](https://mozilla-hub.atlassian.net/browse/FXA-5557). I will be updating that ticket with some questions for Vijay shortly.

Reply:
Ben: This `productId` currently is for Stripe, but this endpoint is for mobile subscriptions. Consider using the existing helpers `fetchSubscriptionsFrom...` and get the corresponding Stripe `productId` from that instead of trying to require the `playSku` or `appStoreProductId` as the identifier in `metricsContext`.